pkg/user/reaper/group_test.go

```diff
@@ -0,0 +1,143 @@
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup", Kind: "Group"}},
+				},
+				&authorizationapi.ClusterRoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-mismatched-subject"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup"}, {Name: "mygroup", Kind: "User"}, {Name: "mygroup", Kind: "Other"}},
+				},
+			},
+			expected: [][]string{
+				{"update", "", "clusterrolebindings", "binding-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+		{
+			name:  "namespaced bindings",
+			group: "mygroup",
+			objects: []runtime.Object{
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-no-subjects", Namespace: "ns1"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{},
+				},
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-one-subject", Namespace: "ns2"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup", Kind: "Group"}},
+				},
+				&authorizationapi.RoleBinding{
+					ObjectMeta: kapi.ObjectMeta{Name: "binding-mismatched-subject", Namespace: "ns3"},
+					RoleRef:    kapi.ObjectReference{Name: "role"},
+					Subjects:   []kapi.ObjectReference{{Name: "mygroup"}, {Name: "mygroup", Kind: "User"}, {Name: "mygroup", Kind: "Other"}},
+				},
+			},
+			expected: [][]string{
+				{"update", "ns2", "rolebindings", "binding-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+		{
+			name:  "sccs",
+			group: "mygroup",
+			objects: []runtime.Object{
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-no-subjects"},
+					Groups:     []string{},
+				},
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-one-subject"},
+					Groups:     []string{"mygroup"},
+				},
+				&kapi.SecurityContextConstraints{
+					ObjectMeta: kapi.ObjectMeta{Name: "scc-mismatched-subjects"},
+					Users:      []string{"mygroup"},
+					Groups:     []string{"mygroup2"},
+				},
+			},
+			expected: [][]string{
+				{"update", "", "securitycontextconstraints", "scc-one-subject"},
+				{"delete", "", "groups", "mygroup"},
+			},
+		},
+	}
+
+	accessor := meta.NewAccessor()
+
+	for _, test := range tests {
+		tc := testclient.NewSimpleFake(test.objects...)
+		ktc := ktestclient.NewSimpleFake(test.objects...)
+
+		actual := [][]string{}
+		reactor := func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
+			switch action := action.(type) {
+			case ktestclient.UpdateAction:
+				obj := action.GetObject()
+				name, err := accessor.Name(obj)
+				if err != nil {
+					t.Fatalf("%s: unexpected error: %v", test.name, err)
+				}
+				actual = append(actual, []string{action.GetVerb(), action.GetNamespace(), action.GetResource(), name})
```

Comment:
because 
```
                       expected: [][]string{
                               {"update", "", "clusterrolebindings", "binding-one-subject"},
                               {"delete", "", "users", "bob"},
                       },
```
was easier to read than
```
			expected: []interface{}{
				ktestclient.UpdateActionImpl{ktestclient.ActionImpl{Verb: "update", Resource: "clusterrolebindings"}, &authorizationapi.ClusterRoleBinding{
					ObjectMeta: kapi.ObjectMeta{Name: "binding-one-subject"},
					RoleRef:    kapi.ObjectReference{Name: "role"},
					Subjects:   []kapi.ObjectReference{},
				}},
				ktestclient.DeleteActionImpl{ktestclient.ActionImpl{Verb: "delete", Resource: "users"}, "bob"},
			},
```